File: Backend/project/apps/financings/clases/type_guarantee.py
import json

class Hipoteca:
    def __init__(self, noEscritura="", notario="", finca="", folio="", libro="", area="", ubicacion="", descripcion="", valor_comercial=0, titular="", estatus="", noContratoArrendamiento="", avaluoBien="", docDigitalSoporte=""):
        self.noEscritura = noEscritura
        self.notario = notario
        self.finca = finca
        self.folio = folio
        self.libro = libro
        self.area = area
        self.ubicacion = ubicacion
        self.descripcion = descripcion
        self.valor_comercial = valor_comercial
        self.titular = titular
        self.estatus = estatus
        self.noContratoArrendamiento = noContratoArrendamiento
        self.avaluoBien = avaluoBien
        self.docDigitalSoporte = docDigitalSoporte
    
    def toJson(self):
        js = {
            'Nombre':'HIPOTECA',
            'NoEscritura':self.noEscritura,
            'Notario':self.notario,
            'Finca':self.finca,
            'Folio':self.folio,
            'Libro':self.libro,
            'Area':self.area,
            'Ubicación':self.ubicacion,
            'Descripción':self.descripcion,
            'Valor Comercial':self.valor_comercial,
            'Titular':self.titular,
            'Estatus':self.estatus,
            'NoContrato_de_Arrendamiento':self.noContratoArrendamiento,
            'Avaluo_del_Bien':self.avaluoBien,
            'Documento_Digital_de_Soporte':self.docDigitalSoporte
        }
        return json.dumps(js, indent=4, ensure_ascii=False)
    
    def __str__(self):
        return f'Hipoteca\nNo.Escritura: {self.noEscritura}'

class DerechoDePosesionHipoteca:
    def __init__(self, noEscritura="", notario="", area="", ubicacion="", descripcion="", valor_comercial=0, titular="", estatus="", noContratoArrendamiento="", avaluoBien="", docDigitalSoporte=""):
        self.noEscritura = noEscritura
        self.notario = notario
        self.area = area
        self.ubicacion = ubicacion
        self.descripcion = descripcion
        self.valor_comercial = valor_comercial
        self.titular = titular
        self.estatus = estatus
        self.noContratoArrendamiento = noContratoArrendamiento
        self.avaluoBien = avaluoBien
        self.docDigitalSoporte = docDigitalSoporte
    
    def toJson(self):
        js = {
            'Nombre':'DERECHO DE POSESIÓN HIPOTECA',
            'NoEscritura':self.noEscritura,
            'Notario':self.notario,           
            'Area':self.area,
            'Ubicacion':self.ubicacion,
            'Descripcion':self.descripcion,
            'Valor_Comercial':self.valor_comercial,
            'Titular':self.titular,
            'Estatus':self.estatus,
            'NoContrato_de_Arrendamiento':self.noContratoArrendamiento,
            'Avaluo_del_Bien':self.avaluoBien,
            'Documento_Digital_de_Soporte':self.docDigitalSoporte
        }
        return json.dumps(js, indent=4, ensure_ascii=False)

class Fiador:
    def __init__(self, codigo_cliente="", nombre="", lugar_trabajo="", ingresos=0, numeroTelefono="", fotografia=""):
        self.codigo_cliente = codigo_cliente
        self.nombre = nombre
        self.lugar_trabajo = lugar_trabajo
        self.ingresos = ingresos
        self.numeroTelefono = numeroTelefono
        self.fotografia = fotografia

    def toJson(self):
        js = {
            'Nombre':'FIADOR',
            'Codigo_de_Cliente':self.codigo_cliente,
            'Nombre':self.nombre,           
            'Lugar de Trabajo':self.lugar_trabajo,
            'Ingresos':self.ingresos,
            'Numero de Telefono':self.descripcion,
            'Fotografia':self.fotografia
        }
        return json.dumps(js, indent=4, ensure_ascii=False)

class Cheque:
    def __init__(self, noCheque="", nombreCuenta="", banco="", cheque_girado_a="", monto_cheque=0, fotografia_cheque=""):
        self.noCheque = noCheque
        self.nombreCuenta = nombreCuenta
        self.banco = banco
        self.cheque_girado_a = cheque_girado_a
        self.monto_cheque = monto_cheque
        self.fotografia_cheque = fotografia_cheque
    
    def toJson(self):
        js = {
            'Nombre':'CHEQUE',
            'noCheque':self.noCheque,
            'NombreCuenta':self.nombreCuenta,
            'Banco':self.banco,
            'Cheque_girado_a':self.cheque_girado_a,
            'Monto_cheque':self.monto_cheque,
            'Fotografia_Cheque':self.fotografia_cheque

        }
        return json.dumps(js, indent=4, ensure_ascii=False)

class Vehiculo:
    def __init__(self, placa="", marca="", color="", noChasis="", noMotor="", valor_comercial=0, fotografias=None, tarjetaCirculacion="", titulo="", noPoliza="", montoSeguro=0, noContratoArrendamiento=""):
        if fotografias is None:
            fotografias = []
        self.placa = placa
        self.marca = marca
        self.color = color
        self.noChasis = noChasis
        self.noMotor = noMotor
        self.valor_comercial = valor_comercial
        self.fotografias = fotografias
        self.tarjetaCirculacion = tarjetaCirculacion
        self.titulo = titulo
        self.noPoliza = noPoliza
        self.montoSeguro = montoSeguro
        self.noContratoArrendamiento = noContratoArrendamiento

    def toJson(self):
        js = {
            'Nombre':'VEHICULO',
            'Placa':self.placa,
            'Marca':self.marca,
            'Color':self.noChasis,
            'NoMotor':self.noMotor,
            'Valor_Comericial':self.valor_comercial,
            'Fotografias':self.fotografias,
            #'Titulo':self.titulo,
            'NoPoliza':self.noPoliza,
            'MontoSeguro':self.montoSeguro,
            'NoContratoArrendamiento':self.noContratoArrendamiento

        }
        return json.dumps(js, indent=4, ensure_ascii=False)

class Mobiliaria:
    def __init__(self, descripcionBien="", documentoAcredita="", imagenDocumentoAcredita="", fotografiaBien="", noPoliza="", montoSeguro=0):
        self.descripcionBien = descripcionBien
        self.documentoAcredita = documentoAcredita
        self.imagenDocumentoAcredita = imagenDocumentoAcredita
        self.fotografiaBien = fotografiaBien
        self.noPoliza = noPoliza
        self.montoSeguro = montoSeguro

    def toJson(self):
        js = {
            'Nombre':'MOBILIARIA',
            'DescripcionBien':self.descripcionBien,
            'DocumentoAcredita':self.documentoAcredita,
            'imagenDocumentoAcredita':self.imagenDocumentoAcredita,
            'FotografiaBien':self.fotografiaBien,
            'NoPoliza':self.noPoliza,
            'MontoSeguro':self.montoSeguro

        }
        return json.dumps(js, indent=4, ensure_ascii=False)

if __name__ == '__main__':
    garantia = Hipoteca()
    print(garantia.toJson())

File: Backend/project/static/js/class/type_guarantee.js
class Hipoteca {
    constructor(noEscritura = "", notario = "", finca = "", folio = "", libro = "", area = "", ubicacion = "", descripcion = "", valor_comercial = 0, titular = "", estatus = "", noContratoArrendamiento = "", avaluoBien = "", docDigitalSoporte = "") {
        this.noEscritura = noEscritura;
        this.notario = notario;
        this.finca = finca;
        this.folio = folio;
        this.libro = libro;
        this.area = area;
        this.ubicacion = ubicacion;
        this.descripcion = descripcion;
        this.valor_comercial = valor_comercial;
        this.titular = titular;
        this.estatus = estatus;
        this.noContratoArrendamiento = noContratoArrendamiento;
        this.avaluoBien = avaluoBien;
        this.docDigitalSoporte = docDigitalSoporte;
    }

    toJson() {
        const js = {
            Nombre: 'HIPOTECA',
            NoEscritura: this.noEscritura,
            Notario: this.notario,
            Finca: this.finca,
            Folio: this.folio,
            Libro: this.libro,
            Area: this.area,
            Ubicación: this.ubicacion,
            Descripción: this.descripcion,
            Valor_Comercial: this.valor_comercial,
            Titular: this.titular,
            Estatus: this.estatus,
            NoContrato_de_Arrendamiento: this.noContratoArrendamiento,
            Avaluo_del_Bien: this.avaluoBien,
            Documento_Digital_de_Soporte: this.docDigitalSoporte
        };
        return JSON.stringify(js, null, 4);
    }

    toString() {
        return `Hipoteca\nNo.Escritura: ${this.noEscritura}`;
    }
}

class DerechoDePosesionHipoteca {
    constructor(noEscritura = "", notario = "", area = "", ubicacion = "", descripcion = "", valor_comercial = 0, titular = "", estatus = "", noContratoArrendamiento = "", avaluoBien = "", docDigitalSoporte = "") {
        this.noEscritura = noEscritura;
        this.notario = notario;
        this.area = area;
        this.ubicacion = ubicacion;
        this.descripcion = descripcion;
        this.valor_comercial = valor_comercial;
        this.titular = titular;
        this.estatus = estatus;
        this.noContratoArrendamiento = noContratoArrendamiento;
        this.avaluoBien = avaluoBien;
        this.docDigitalSoporte = docDigitalSoporte;
    }

    toJson() {
        const js = {
            Nombre: 'DERECHO DE POSESIÓN HIPOTECA',
            NoEscritura: this.noEscritura,
            Notario: this.notario,
            Area: this.area,
            Ubicacion: this.ubicacion,
            Descripcion: this.descripcion,
            Valor_Comercial: this.valor_comercial,
            Titular: this.titular,
            Estatus: this.estatus,
            NoContrato_de_Arrendamiento: this.noContratoArrendamiento,
            Avaluo_del_Bien: this.avaluoBien,
            Documento_Digital_de_Soporte: this.docDigitalSoporte
        };
        return JSON.stringify(js, null, 4);
    }
}

class Fiador {
    constructor(codigo_cliente = "", nombre = "", lugar_trabajo = "", ingresos = 0, numeroTelefono = "", fotografia = "") {
        this.codigo_cliente = codigo_cliente;
        this.nombre = nombre;
        this.lugar_trabajo = lugar_trabajo;
        this.ingresos = ingresos;
        this.numeroTelefono = numeroTelefono;
        this.fotografia = fotografia;
    }

    toJson() {
        const js = {
            Nombre: 'FIADOR',
            Codigo_de_Cliente: this.codigo_cliente,
            Nombre: this.nombre,
            Lugar_de_Trabajo: this.lugar_trabajo,
            Ingresos: this.ingresos,
            Numero_de_Telefono: this.numeroTelefono,
            Fotografia: this.fotografia
        };
        return JSON.stringify(js, null, 4);
    }
}

class Cheque {
    constructor(noCheque = "", nombreCuenta = "", banco = "", cheque_girado_a = "", monto_cheque = 0, fotografia_cheque = "") {
        this.noCheque = noCheque;
        this.nombreCuenta = nombreCuenta;
        this.banco = banco;
        this.cheque_girado_a = cheque_girado_a;
        this.monto_cheque = monto_cheque;
        this.fotografia_cheque = fotografia_cheque;
    }

    toJson() {
        const js = {
            Nombre: 'CHEQUE',
            noCheque: this.noCheque,
            NombreCuenta: this.nombreCuenta,
            Banco: this.banco,
            Cheque_girado_a: this.cheque_girado_a,
            Monto_cheque: this.monto_cheque,
            Fotografia_Cheque: this.fotografia_cheque
        };
        return JSON.stringify(js, null, 4);
    }
}

class Vehiculo {
    constructor(placa = "", marca = "", color = "", noChasis = "", noMotor = "", valor_comercial = 0, fotografias = [], tarjetaCirculacion = "", titulo = "", noPoliza = "", montoSeguro = 0, noContratoArrendamiento = "") {
        this.placa = placa;
        this.marca = marca;
        this.color = color;
        this.noChasis = noChasis;
        this.noMotor = noMotor;
        this.valor_comercial = valor_comercial;
        this.fotografias = fotografias;
        this.tarjetaCirculacion = tarjetaCirculacion;
        this.titulo = titulo;
        this.noPoliza = noPoliza;
        this.montoSeguro = montoSeguro;
        this.noContratoArrendamiento = noContratoArrendamiento;
    }

    toJson() {
        const js = {
            Nombre: 'VEHICULO',
            Placa: this.placa,
            Marca: this.marca,
            Color: this.color,
            NoChasis: this.noChasis,
            NoMotor: this.noMotor,
            Valor_Comercial: this.valor_comercial,
            Fotografias: this.fotografias,
            TarjetaCirculacion: this.tarjetaCirculacion,
            Titulo: this.titulo,
            NoPoliza: this.noPoliza,
            MontoSeguro: this.montoSeguro,
            NoContratoArrendamiento: this.noContratoArrendamiento
        };
        return JSON.stringify(js, null, 4);
    }
}

class Mobiliaria {
    constructor(descripcionBien = "", documentoAcredita = "", imagenDocumentoAcredita = "", fotografiaBien = "", noPoliza = "", montoSeguro = 0) {
        this.descripcionBien = descripcionBien;
        this.documentoAcredita = documentoAcredita;
        this.imagenDocumentoAcredita = imagenDocumentoAcredita;
        this.fotografiaBien = fotografiaBien;
        this.noPoliza = noPoliza;
        this.montoSeguro = montoSeguro;
    }

    toJson() {
        const js = {
            Nombre: 'MOBILIARIA',
            DescripcionBien: this.descripcionBien,
            DocumentoAcredita: this.documentoAcredita,
            ImagenDocumentoAcredita: this.imagenDocumentoAcredita,
            FotografiaBien: this.fotografiaBien,
            NoPoliza: this.noPoliza,
            MontoSeguro: this.montoSeguro
        };
        return JSON.stringify(js, null, 4);
    }
}
